crypto/signer.go

```diff
@@ -69,28 +66,14 @@ func (s *Signer) Sign(data []byte) (*sigpb.DigitallySigned, error) {
 	}, nil
 }
 
-// SignObject signs the requested object using ObjectHash.
-func (s *Signer) SignObject(obj interface{}) (*sigpb.DigitallySigned, error) {
-	// TODO(gbelvin): use objecthash.CommonJSONify
-	j, err := json.Marshal(obj)
-	if err != nil {
-		return nil, err
-	}
-	hash, err := objecthash.CommonJSONHash(string(j))
-	if err != nil {
-		return nil, fmt.Errorf("CommonJSONHash(%s): %v", j, err)
-	}
-	return s.Sign(hash[:])
-}
-
 // SignLogRoot hashes and signs the supplied (to-be) SignedLogRoot and returns a
 // signature.  Hashing is performed by github.com/benlaurie/objecthash.
 func (s *Signer) SignLogRoot(root *trillian.SignedLogRoot) (*sigpb.DigitallySigned, error) {
-	hash, err := HashLogRoot(*root)
+	canonical, err := CanonicalLogRoot(root, LogRootV0)
 	if err != nil {
 		return nil, err
 	}
-	signature, err := s.Sign(hash)
+	signature, err := s.Sign(canonical)
```

Comment:
[As a complete aside, I think it would be nice to rename `trillian.crypto`'s `Signer.Sign()` method to something like `HashAndSign()` so it a) better indicates what it does and b) distinguishes it from `crypto`'s `Signer.Sign()` method...]

Reply:
Good idea. How about `SignData`?  Technically `signer.Sign` signs a hash...